conan/cli/commands/upload.py

```diff
@@ -101,8 +101,8 @@ def upload(conan_api: ConanAPI, parser, *args):
         if not args.list and not args.confirm and "*" in args.pattern:
             _ask_confirm_upload(conan_api, package_list)
 
-        conan_api.upload.upload_full(package_list, remote, enabled_remotes, args.check,
-                                     args.force, args.metadata, args.dry_run)
+        package_list = conan_api.upload.upload_full(package_list, remote, enabled_remotes, args.check,
```

Comment:
I think it will be even simpler by adding it in-place in the ``package_list`` argument, instead of duplicating and appending to it. Note users have the power to either copy it in advance, and commands won't change an existing file, unless users explicitly tell Conan to do so.

Reply:
I'm going to give it a try, I was worried about this because of the upload thread pool and the package_list split.

`thread_pool.map(_upload_pkglist, package_list.split())`